packages/table/src/quadrants/tableQuadrantStack.tsx

```diff
@@ -0,0 +1,438 @@
+                    renderRowHeader={this.renderMainQuadrantRowHeader}
+                    scrollContainerRef={this.quadrantRefHandlers[QuadrantType.MAIN].scrollContainer}
+                />
+                <TableQuadrant
+                    grid={grid}
+                    isRowHeaderShown={isRowHeaderShown}
+                    onWheel={this.handleTopQuadrantWheel}
+                    quadrantRef={this.quadrantRefHandlers[QuadrantType.TOP].quadrant}
+                    quadrantType={QuadrantType.TOP}
+                    renderBody={renderBody}
+                    renderColumnHeader={renderColumnHeader}
+                    renderMenu={this.renderTopQuadrantMenu}
+                    renderRowHeader={this.renderTopQuadrantRowHeader}
+                    scrollContainerRef={this.quadrantRefHandlers[QuadrantType.TOP].scrollContainer}
+                />
+                <TableQuadrant
+                    grid={grid}
+                    isRowHeaderShown={isRowHeaderShown}
+                    onWheel={this.handleLeftQuadrantWheel}
+                    quadrantRef={this.quadrantRefHandlers[QuadrantType.LEFT].quadrant}
+                    quadrantType={QuadrantType.LEFT}
+                    renderBody={renderBody}
+                    renderColumnHeader={renderColumnHeader}
+                    renderMenu={this.renderLeftQuadrantMenu}
+                    renderRowHeader={this.renderLeftQuadrantRowHeader}
+                    scrollContainerRef={this.quadrantRefHandlers[QuadrantType.LEFT].scrollContainer}
+                />
+                <TableQuadrant
+                    grid={grid}
+                    isRowHeaderShown={isRowHeaderShown}
+                    onWheel={this.handleTopLeftQuadrantWheel}
+                    quadrantRef={this.quadrantRefHandlers[QuadrantType.TOP_LEFT].quadrant}
+                    quadrantType={QuadrantType.TOP_LEFT}
+                    renderBody={renderBody}
+                    renderColumnHeader={renderColumnHeader}
+                    renderMenu={this.renderTopLeftQuadrantMenu}
+                    renderRowHeader={this.renderTopLeftQuadrantRowHeader}
+                    scrollContainerRef={this.quadrantRefHandlers[QuadrantType.TOP_LEFT].scrollContainer}
+                />
+            </div>
+        );
+    }
+
+    // Ref handlers
+    // ============
+
+    private generateQuadrantRefHandlers(quadrantType: QuadrantType): IQuadrantRefHandlers {
+        const reducer = (agg: IQuadrantRefHandlers, key: keyof IQuadrantRefHandlers) => {
+            agg[key] = (ref: HTMLElement) => {
+                this.quadrantRefs[quadrantType][key] = ref;
+                if (quadrantType === QuadrantType.MAIN) {
+                    if (key === "quadrant") {
+                        CoreUtils.safeInvoke(this.props.quadrantRef, ref);
+                    } else if (key === "scrollContainer") {
+                        CoreUtils.safeInvoke(this.props.scrollContainerRef, ref);
+                    }
```

Comment:
This area feels weird too. I want to keep a private `ref` to each of these things but _also_ emit the `ref` via the provided props. Is there a better way to do this while also keeping this generator function around? Do we need the generator function?